server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -544,8 +545,13 @@ private void startElection() {
             // The preVoteCollector is only active while we are candidate, but it does not call this method with synchronisation, so we have
             // to check our mode again here.
             if (mode == Mode.CANDIDATE) {
-                if (localNodeMayWinElection(getLastAcceptedState(), electionStrategy) == false) {
-                    logger.trace("skip election as local node may not win it: {}", getLastAcceptedState().coordinationMetadata());
+                final var nodeEligibility = localNodeMayWinElection(getLastAcceptedState(), electionStrategy);
+                if (nodeEligibility.mayWin() == false) {
+                    logger.info(
```

Comment:
we're changing what used to be a TRACE to info. Are you sure that it doesn't produce too much noise? Why not change the log level depending on whether there is a reason or not?

Reply:
Ah, I see. I believe we're doing what you suggest -- there will always be a non-empty `reason` set if `mayWin=false`. Confused me for a moment, too.

As to logging at INFO instead of TRACE, hmm... I'm not sure what the contract here is, for how often this code runs in steady state. If this code doesn't run in steady state, then that'd be okay. But if we're constantly checking and logging, then that would be bad.

Reply:
I think I've tracked this down. My conclusion is that the logs can be quite chatty when there is no master + the node is not allowed to run for election.

We have a default [gracePeriod val of zero](https://github.com/elastic/elasticsearch/blob/31c05e9528772731aa64c8558c25ab8ef1165d51/server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java#L1763), and then we use [these settings](https://github.com/elastic/elasticsearch/blob/31c05e9528772731aa64c8558c25ab8ef1165d51/server/src/main/java/org/elasticsearch/cluster/coordination/ElectionSchedulerFactory.java#L61-L91), which are on the order of milliseconds, to [semi-randomly pick a delay](https://github.com/elastic/elasticsearch/blob/31c05e9528772731aa64c8558c25ab8ef1165d51/server/src/main/java/org/elasticsearch/cluster/coordination/ElectionSchedulerFactory.java#L177) for scheduling runs to try for election.

I think the question boils down to whether we want the new logging for shutdown to be at INFO or TRACE. My preference is consistency in logging the reason, whether the reason be due to shutdown or something else.